src/core/public/application/application_service.tsx

```diff
@@ -242,11 +242,17 @@ export class ApplicationService {
       appId,
       { path, state, replace = false }: NavigateToAppOptions = {}
     ) => {
-      if (await this.shouldNavigate(overlays)) {
+      const currentAppId = this.currentAppId$.value;
+      const navigatingToSameApp = currentAppId === appId;
+      const shouldNavigate = navigatingToSameApp ? true : await this.shouldNavigate(overlays);
```

Comment:
KP Route doesn't know about plugin internal routes, why we use it to perform navigations inside an app?

Reply:
Well, apps can/are using this API to perform intra-app nanavigation, with utilities such as `RedirectAppLinks`, and overall all usages of `navigateToApp` or `navigateToUrl` used to navigate within your own app.

Reply:
could you add a comment why we don't call `shouldNavigate`? it's not obvious that `shouldNavigate` calls confirmation modal under the hood.